src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/AutomatedAcceptanceTestsSetup.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.cmc.claimstore;
+
+import io.restassured.RestAssured;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.aat.TestInstance;
+import uk.gov.hmcts.cmc.claimstore.aat.idam.IdamTestService;
+import uk.gov.hmcts.cmc.claimstore.aat.idam.TestUser;
+import uk.gov.hmcts.cmc.claimstore.idam.IdamApi;
+
+import javax.annotation.PostConstruct;
+
+@Primary
+@Component
+@Profile("aat-tests")
+public class AutomatedAcceptanceTestsSetup implements TestsSetup {
+
+    private final IdamTestService idamTestService;
+    private final IdamApi idamApi;
+    private final TestUser testUser;
+    private final TestInstance testInstance;
+
+    private String authenticationToken;
+    private String userId;
+
+    @Autowired
+    public AutomatedAcceptanceTestsSetup(
+        IdamTestService idamTestService,
+        IdamApi idamApi,
+        TestUser testUser,
+        TestInstance testInstance
+    ) {
+        this.idamTestService = idamTestService;
+        this.idamApi = idamApi;
+        this.testUser = testUser;
+        this.testInstance = testInstance;
+    }
+
+    @PostConstruct
+    public void initialize() {
+        RestAssured.baseURI = testInstance.getUri();
+        authenticationToken = idamTestService.logIn(testUser.getUsername(), testUser.getPassword());
+        userId = idamApi.retrieveUserDetails("Bearer " + authenticationToken).getId();
```

Comment:
I wonder whether we should just get the id out of the JWT in our tests?

Reply:
One call less, I will try that 👍